### DISCUSSION TITLE
EntranceNode和DefaultNode构建的invocation tree ，会一直存在内存中吗 ，会定期清理吗？

### DISCUSSION THREAD

**Discussion by A491382** [alibaba/Sentinel#2097] [Q&A]
EntranceNode和DefaultNode构建的invocation tree 会一直存在内存中吗 ？ 还是会有定期清理，如果一直存在内存中是否会有最大值之类的控制呢 ？

**Comment by sczyh30** [comment]
目前不会进行清理，为防止数目爆炸 会限制总的数目。欢迎社区提交 PR 完善这一块

---

EntranceNode and DefaultNode won't be evicted.

**Reply by A491382** [reply]
明白了， 非常感谢

**Comment by zidoshare** [comment]
@sczyh30 这一块虽然没有那么重要，但是是否可以采用这种优化
* 当数额未超过MAX_CONTEXT_NAME_SIZE，正常处理
* 当数额超过MAX_CONTEXT_NAME_SIZE，清理 intervalInMs 内无记录的node让出空间。

另外MAX_CONTEXT_NAME_SIZE也可以考虑提供出来供使用者调节。

这样处理在正常情况下应该对性能不会有什么影响，因为一般没有那么多资源限流。而如果真正遇到了需要限流的资源太多，也能够进行处理，而不是直接报错。不过那种场景可能会导致频繁new EntranceNode。而这也可以通过调节MAX_CONTEXT_NAME_SIZE来达到平衡。

**Reply by sczyh30** [reply]
Good idea. 欢迎提交 PR 来完善这一块~
